tensorflow/lite/experimental/micro/kernels/cmsis-nn/pooling.cc

```diff
@@ -0,0 +1,278 @@
+  CalculateActivationRangeUint8(params->activation, output, &activation_min,
+                                &activation_max);
+
+  PoolParams op_params;
+  op_params.stride_height = params->stride_height;
+  op_params.stride_width = params->stride_width;
+  op_params.filter_height = params->filter_height;
+  op_params.filter_width = params->filter_width;
+  op_params.padding_values.height = data->padding.height;
+  op_params.padding_values.width = data->padding.width;
+  op_params.quantized_activation_min = activation_min;
+  op_params.quantized_activation_max = activation_max;
+  reference_ops::AveragePool(
+      op_params, GetTensorShape(input), GetTensorData<uint8_t>(input),
+      GetTensorShape(output), GetTensorData<uint8_t>(output));
+}
+
+TfLiteStatus AverageEvalInt8(const TfLiteContext* context,
+                             const TfLiteNode* node,
+                             const TfLitePoolParams* params, const OpData* data,
+                             const TfLiteTensor* input, TfLiteTensor* output) {
+  int32_t activation_min, activation_max;
+  CalculateActivationRangeInt8(params->activation, output, &activation_min,
+                               &activation_max);
+
+  TFLITE_DCHECK_LE(params->quantized_activation_min,
+                   params->quantized_activation_max);
+
+  RuntimeShape input_shape = GetTensorShape(input);
+  TFLITE_DCHECK_EQ(input_shape.DimensionsCount(), 4);
+
+  RuntimeShape output_shape = GetTensorShape(output);
+  TFLITE_DCHECK_EQ(output_shape.DimensionsCount(), 4);
+
+  const int batches = MatchingDim(input_shape, 0, output_shape, 0);
+  const int depth = MatchingDim(input_shape, 3, output_shape, 3);
+  const int input_height = input_shape.Dims(1);
+  const int input_width = input_shape.Dims(2);
+  const int output_height = output_shape.Dims(1);
+  const int output_width = output_shape.Dims(2);
+  const int stride_height = params->stride_height;
+  const int stride_width = params->stride_width;
+
+  const int filter_height = params->filter_height;
+  const int filter_width = params->filter_width;
+  const int padding_height = data->padding.height;
+  const int padding_width = data->padding.width;
+
+#if defined(ARM_MATH_DSP) && defined(ARM_MATH_LOOPUNROLL)
+  int16_t* scratch_buffer = nullptr;
+  const int32_t buffer_size =
+      arm_avgpool_s8_get_buffer_size(output_width, depth);
+
+  TF_LITE_ENSURE_OK(
+      context, get_cmsis_scratch_buffer(context, &scratch_buffer, buffer_size));
+
+  TF_LITE_ENSURE_EQ(
+      context,
+      arm_avgpool_s8(input_height, input_width, output_height, output_width,
+                     stride_height, stride_width, filter_height, filter_width,
+                     padding_height, padding_width, activation_min,
+                     activation_max, depth, GetTensorData<int8_t>(input),
+                     scratch_buffer, GetTensorData<int8_t>(output)),
+      ARM_MATH_SUCCESS);
+#else
+#error ARM_MATH_DSP and ARM_MATH_LOOPUNROLL must be set
```

Comment:
I know this is what we do for the other opt kernels. However, we're going to change this pattern to fall back to the reference kernel implementation. So something like:

#else
  tflite::reference_ops::AveragePool(...)
#endif

It would be great if you added that!

Reply:
Thanks for the info!
I have made the necessary changes